### PR TITLE
HDDS-5560. Updating clusterId to correct value

### PR DESCRIPTION
**Pull request by nishitpatira**
## What changes were proposed in this pull request?

ClusterId was being incorrectly set as datanodeId in the dnlayoutversion/version file.
While this field was not being used anywhere and poses no immediate problem, we wanted to fix it to reflect the correct value.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-5560

## How was this patch tested?

Built successfully.
Local test on docker to see the value is being updated correctly.

### CODE REVIEW

**Comment by nishitpatira** [comment]
@neils-dev @errose28 Can we please review this?

**Comment by neils-dev** [comment]
Thanks @nishitpatira for submitting this PR for fixing the dnlayoutversion file to properly container the clusterID.  I;ve applied your patch and brought up the dev docker cluster.  I verified the clusterID for the dnlayoutversion file is the same as the the clusterID marked in the hdds VERSION file as follows,
```
$ docker-compose exec datanode bash
$ cat /data/metadata/dnlayoutversion/VERSION
$ cat /data/hdds/hdds/VERSION

```
Would be great if you would update similar steps to test under the "How patch was tested section".  Thanks.

**Comment by neils-dev** [comment]
Is there a way we can verify if this update to the `dnlayoutversion` file does not affect older versions of ozone, 2.1, 1?  @errose28 is this something we need to do?

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/DatanodeLayoutStorage.java [hunk 1]
```diff
@@ -64,6 +64,11 @@ protected Properties getNodeProperties() {
     return new Properties();
   }
 
+  @Override
+  public void setClusterId(String clusterId) throws IOException {
+    super.getStorageInfo().setClusterId(clusterId);
```
Comment (neils-dev) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/DatanodeLayoutStorage.java:69]: Thanks @nishitpatira for this patch.  We don't need to override the `setClusterId` method here.  There is already an implementation provided by the `abstract Storage class`.  Not needed here, just use the base class implementation.

**Review by neils-dev — COMMENTED** [review]
A test for the setting of the storage version file with the `datanode` `clusterID` is a good idea.  A test similar(ish) to `TestEndPoint.testCheckVersionResponse`,
https://github.com/apache/ozone/blob/0a377c685ad8ee6ebb35ef40d5ad1f709fb590e1/hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/ozone/container/common/TestEndPoint.java#L169

checking that the version file contains the `datanode` `clusterId` will suffice.

**Comment by DaveTeng0** [comment]
Hey @nishitpatira ~ there were some comments from the reviewer for this PR. Please help take a look again when you have time, thanks!

**Comment by adoroszlai** [comment]
This PR is waiting for a test case to be added.  Instead of marking it as `pending` and getting it auto-closed, I'd like to ask if @DaveTeng0 or @neils-dev would like to contribute the test case?

**Comment by neils-dev** [comment]
Thanks @adoroszlai for tending to the status of this PR.  I've gone ahead and implemented a test for it.  The PR patch however needs a bit of work, upon closer look I found an exception raised repeatedly with the current impl.  
```
ozone-datanode-1  | java.io.IOException: Storage directory /data/metadata/dnlayoutversion already initialized.
ozone-datanode-1  | 	at org.apache.hadoop.ozone.common.Storage.setClusterId(Storage.java:123)
ozone-datanode-1  | 	at org.apache.hadoop.ozone.container.common.states.endpoint.VersionEndpointTask.call(VersionEndpointTask.java:81)
ozone-datanode-1  | 	at org.apache.hadoop.ozone.container.common.states.endpoint.VersionEndpointTask.call(VersionEndpointTask.java:41)
ozone-datanode-1  | 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
ozone-datanode-1  | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
ozone-datanode-1  | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
ozone-datanode-1  | 	at java.base/java.lang.Thread.run(Thread.java:829)
ozone-datanode-3  | 2023-03-09 03:41:28,013 [EndpointStateMachine task thread for scm/172.20.0.5:9861 - 0 ] WARN statemachine.EndpointStateMachine: Unable to communicate to SCM server at scm:9861 for past 350 seconds.
```

Needs a little work.  I'll push an update when resolved.

**Comment by neils-dev** [comment]
thanks @adoroszlai .  Created a unit test, `TestEndPoint#testDnLayoutVersionFile`.  Currently in my branch linked to this PR, see 
https://github.com/apache/ozone/compare/master...neils-dev:ozone:pr/4110.
Pls take a look and we can add it.

**Comment by adoroszlai** [comment]
@DaveTeng0 can you please review @neils-dev's updates in https://github.com/apache/ozone/commit/63cdd7c4dafe7e0f1f95b8b555e23f51db47001e?

**Comment by adoroszlai** [comment]
@sumitagrawl can you please review this?

**Comment by adoroszlai** [comment]
@neils-dev thanks for the unit test, I've added some comments on 63cdd7c4dafe7e0f1f95b8b555e23f51db47001e.

**Comment by ashishkumar50** [comment]
@neils-dev @nishitpatira If test case is added Can you please merge and close this PR.

**Comment by adoroszlai** [comment]
Continued in #5289.

**Pull request closed without merging**